### DISCUSSION TITLE
Providing TTFont to fontTools.merge?

### DISCUSSION THREAD

**Discussion by NightFurySL2001** [fonttools/fonttools#3614] [General]
Is it currently possible to provide TTFont object instead of font paths into fontTools.merge.Merger?

**Comment by anthrotype** [comment]
apparently not, but it shouldn't be too difficult to modify it so that it works with either paths or TTFonts.
PR most welcome!

**Reply by NightFurySL2001** [reply]
I see. Merger seem to need more works though compared to [`ufomerge`](https://github.com/googlefonts/ufomerge), mainly on `--skip-existing`/`--replace-existing`, not sure if I have the time to mesh it in.

**Comment by ftCLI** [comment]
If you have already initilized TTFont objects and they have been loaded from files, you can pass `TTFont.reader.file.name` to the Merger. Otherwise the `Merger` class needs to be modified.

**Reply by anthrotype** [reply]
> you can pass TTFont.reader.file.name to the Merger.

but then the Merger will attempt to load new objects from those same filenames, it won't modify the existing objects. I don't think that's what the OP wanted

**Reply by ftCLI** [reply]
Yep. It was a quick and dirty workaround (I don't know what's he's coding). But as you told, modifying the Merger shouldn't be difficult.

**Reply by NightFurySL2001** [reply]
I'm planning for a multistep flow which involves resizing UPM with [`otRebuilder`](https://github.com/Pal3love/otRebuilder/blob/master/Package/otRebuilder/Lib/Converter.py) and it operates on TTFont object. Later the TTFont is passed into Merger to quickly include missing glyphs.
